mantidproject/mantid
# Re #20073: Modified GenericDataProcessor to pass message as QString rather than std::exception.

- The reductionError signal has been modified to send a `QString` containing the message from the caught exception rather than a `std::exception` by value which caused object slicing.
- The `GenericDataProcessor` widget (and as a result also the `ReflDataProcessorWidget`) has been modified to use `QString` rather than `std::string`.

**To test:**
1. Open the Reflectometry GUI.
2. Press the process button
3. Confirm the action.
4. Ensure an appropriate error message is shown.

1. Compile and run the unit test targets `CustomInterfacesTest` and `MantidWidgetsTest`.

Fixes #20073

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
